Installation issue: ROOT

### Steps to reproduce the issue

```console
$ export MACOSX_DEPLOYMENT_TARGET=11
$ spack install root
```
eventually while building `core/G__Core` gives:
```
error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11'
error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11'
Error: Parsing Linkdef file /var/folders/gy/mrg1ffts2h945qj9k29s1l1dvvmbqb/T/s3j/spack-stage/spack-stage-root-6.24.06-n52av2vlzcxyfgukqznvjxfjns4j7dx5/spack-src/core/base/inc/LinkDef.h
make[2]: *** [core/G__Core.cxx] Error 1
make[1]: *** [core/CMakeFiles/G__Core.dir/all] Error 2
```

Clearing the `MACOSX_DEPLOYMENT_TARGET` environment variable fixes the problem. The value of `11` works fine for the other compilers on my system, just apparently not RootCling. Perhaps this is because ROOT uses an older forked version of LLVM.

### Information on your system

* **Spack:** 0.17.0-604-c68652c88c
* **Python:** 3.8.5
* **Platform:** darwin-bigsur-skylake
* **Concretizer:** clingo

### Additional information

@HadrienG2  @chissg @drbenmorgan @vvolkl @pcanal

[spack-build-01-cmake-out.txt](https://github.com/spack/spack/files/7777596/spack-build-01-cmake-out.txt)
[spack-build-02-build-out.txt](https://github.com/spack/spack/files/7777597/spack-build-02-build-out.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/root/package.py
+++ b/var/spack/repos/builtin/packages/root/package.py
@@ -605,6 +605,11 @@ class Root(CMakePackage):
         if '+opengl' in spec:
             add_include_path('glew')
             add_include_path('mesa-glu')
+        if 'platform=darwin' in spec:
+            # Newer deployment targets cause fatal errors in rootcling, so
+            # override with an empty value even though it may lead to link
+            # warnings when building against ROOT
+            env.unset('MACOSX_DEPLOYMENT_TARGET')
 
     def setup_run_environment(self, env):
         env.set('ROOTSYS', self.prefix)

